blockchain/My-Wallet-V3-iOS
# feat(KYC): IOS-1104 KYC Authentication

## Objective

Implementing KYC authentication.

## Description

Branched off of the work started by @woudini in PR #398 to finish KYC authentication. There are still a few polish items but I'd like to get this in first and will work on those improvements in another PR.

Summary of changes:
* KYCAuthenticationService provides components with a `KYCSessionTokenResponse` (contains the token that should be added to the HTTP header for all KYC-related network calls) via `getKycSessionToken()`. Currently invoking that method will _always_ retrieve a new session token.
* Updated `BlockchainDataRepository` to use `BehaviorRelay` instead of `Variable` since `Variable` will be a deprecated.

Next PR will address:
* caching the KYC session token (I think it should be closely tied to the lifecycle of `KYCCoordinator`)
* pass in the KYC session token in other KYC network calls
* unit tests

## How to Test

1. Since this includes changes in the `My-Wallet-V3` Submodule, you'll need to make sure to pull in the changes by running `git submodule update` followed by running `sh scripts/install-js.sh && sh scripts/build-js.sh`
2. The KYC flow now requires a valid wallet GUID before proceeding so you'll need to make sure to launch KYC from the debug view after authenticating into your wallet. Do access that go to "Settings" -> "Terms of Service" -> "Debug" -> "Launch KYC"
3. Upon launching KYC, notice that `KYCUser` can now be retrieved

## Screenshot

N/A

## Related Information

* Ticket Number: IOS-1104

## Merge Checklist
